[Bug]: CI test failing at `buildtest buildspec find invalid`

### Bug Description

see https://github.com/buildtesters/buildtest/runs/7794092682?check_suite_focus=true 

Problem is that we now get `buildtest bc find invalid` to raise SystemExit and this causes CI check to fail so we need to fix this 

The regression test is in 

https://github.com/buildtesters/buildtest/blob/3ead042cbb987d7854afa17ef187159c4ec5a35a/tests/cli/test_buildspec.py#L145-L152

```
            # return a non-zero returncode
            if table.row_count > 0:
                console.print(table)
>               sys.exit(1)
E               SystemExit: 1

```

### Version

devel

### Relevant log output

_No response_

### Post question in Slack

- [X] I agree that I posted my question in [slack](https://hpcbuildtest.slack.com/) before creating this issue

### Is there an existing issue

- [X] I confirm there is no existing [issue](https://github.com/buildtesters/buildtest/issues) for this issue

--- a/buildtest/cli/buildspec.py
+++ b/buildtest/cli/buildspec.py
@@ -874,6 +874,7 @@ class BuildspecCache:
 
         if error and terse:
             console.print("The --terse flag can not be used with the --error option")
+            sys.exit(1)
             return
 
         # implementation for machine readable format specified via --terse

--- a/buildtest/cli/buildspec.py
+++ b/buildtest/cli/buildspec.py
@@ -882,6 +882,7 @@ class BuildspecCache:
                 print("buildspec")
             for buildspec in self.cache["invalids"].keys():
                 print(buildspec)
+            sys.exit(1)
             return
 
         # if --error is not specified print list of invalid buildspecs in rich table
@@ -907,6 +908,7 @@ class BuildspecCache:
         for buildspec, value in self.cache["invalids"].items():
             console.rule(buildspec)
             pprint(value)
+        sys.exit(1)
 
     @staticmethod
     def print_filter_fields():

--- a/buildtest/cli/buildspec.py
+++ b/buildtest/cli/buildspec.py
@@ -875,7 +875,6 @@ class BuildspecCache:
         if error and terse:
             console.print("The --terse flag can not be used with the --error option")
             sys.exit(1)
-            return
 
         # implementation for machine readable format specified via --terse
         if terse:
@@ -884,7 +883,6 @@ class BuildspecCache:
             for buildspec in self.cache["invalids"].keys():
                 print(buildspec)
             sys.exit(1)
-            return
 
         # if --error is not specified print list of invalid buildspecs in rich table
         if not error:

--- a/tests/cli/test_buildspec.py
+++ b/tests/cli/test_buildspec.py
@@ -148,10 +148,20 @@ def test_buildspec_find_invalid():
     # testing buildtest buildspec find invalid. This will assert SystemExit exception raised by sys.exit
     with pytest.raises(SystemExit):
         cache.print_invalid_buildspecs(error=True)
+
+    with pytest.raises(SystemExit):
         cache.print_invalid_buildspecs(error=False)
+
+    with pytest.raises(SystemExit):
         cache.print_invalid_buildspecs(error=True, terse=True)
+
+    with pytest.raises(SystemExit):
         cache.print_invalid_buildspecs(error=False, terse=True)
+
+    with pytest.raises(SystemExit):
         cache.print_invalid_buildspecs(error=True, terse=True, header=True)
+
+    with pytest.raises(SystemExit):
         cache.print_invalid_buildspecs(error=False, terse=True, header=True)
 
 

--- a/tests/cli/test_buildspec.py
+++ b/tests/cli/test_buildspec.py
@@ -144,17 +144,15 @@ def test_buildspec_maintainers():
 @pytest.mark.cli
 def test_buildspec_find_invalid():
     cache = BuildspecCache(configuration=configuration)
-    cache.print_invalid_buildspecs(error=True)
 
-    # testing buildtest buildspec find invalid. This will catch SystemExit exception raised by sys.exit
-    with pytest.raises(SystemExit) as pytest_wrapped_e:
+    # testing buildtest buildspec find invalid. This will assert SystemExit exception raised by sys.exit
+    with pytest.raises(SystemExit):
+        cache.print_invalid_buildspecs(error=True)
         cache.print_invalid_buildspecs(error=False)
-    assert pytest_wrapped_e.type == SystemExit
-
-    cache.print_invalid_buildspecs(error=True, terse=True)
-    cache.print_invalid_buildspecs(error=False, terse=True)
-    cache.print_invalid_buildspecs(error=True, terse=True, header=True)
-    cache.print_invalid_buildspecs(error=False, terse=True, header=True)
+        cache.print_invalid_buildspecs(error=True, terse=True)
+        cache.print_invalid_buildspecs(error=False, terse=True)
+        cache.print_invalid_buildspecs(error=True, terse=True, header=True)
+        cache.print_invalid_buildspecs(error=False, terse=True, header=True)
 
 
 @pytest.mark.cli

--- a/tests/cli/test_buildspec.py
+++ b/tests/cli/test_buildspec.py
@@ -145,7 +145,12 @@ def test_buildspec_maintainers():
 def test_buildspec_find_invalid():
     cache = BuildspecCache(configuration=configuration)
     cache.print_invalid_buildspecs(error=True)
-    cache.print_invalid_buildspecs(error=False)
+
+    # testing buildtest buildspec find invalid. This will catch SystemExit exception raised by sys.exit
+    with pytest.raises(SystemExit) as pytest_wrapped_e:
+        cache.print_invalid_buildspecs(error=False)
+    assert pytest_wrapped_e.type == SystemExit
+
     cache.print_invalid_buildspecs(error=True, terse=True)
     cache.print_invalid_buildspecs(error=False, terse=True)
     cache.print_invalid_buildspecs(error=True, terse=True, header=True)

